File: spring/file-upload/src/main/java/com/example/fileupload/MyController.java
package com.example.fileupload;

import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestMapping;

@Controller
@RequestMapping("/")
public class MyController {

    @GetMapping
    public String index() {
        return "index";
    }

    @PostMapping
    public String post() {
        return "post";
    }
}

File: spring/jackson-json-date-type-convert/src/main/kotlin/com/example/jackson_json_date_type_convert/MyController.kt
package com.example.jackson_json_date_type_convert

import org.springframework.stereotype.Controller
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.ResponseBody
import java.time.LocalDate

@Controller
class MyController {

    @GetMapping
    fun get(myForm: MyForm): String {
        return "index"
    }

    @PostMapping
    fun post(myForm: MyForm): String {
        println(myForm.toString())
        return "index"
    }

    @GetMapping("/json")
    @ResponseBody
    fun json(): MyForm {
        return MyForm(
            date1 = LocalDate.of(2021, 1, 1),
            date2 = null,
        )
    }
}
